### DISCUSSION TITLE
Use of framer motion to move poker card into pile

### DISCUSSION THREAD

**Discussion by zhibin-wang09** [aod/zhithead#60] [Q&A] [answered]
Hi!

I wanted to create a poker game using framer motion as well. I looked through your code but still confused on how you used framer motion to move the poker card from hand to pile. Can you provide any insight on how you approached that problem?

**Comment by aod** [comment, marked as answer]
Hi, appreciate your interest! It's been quite a while since I've last wrote code for this project. But I remember it being quite simple. I took a look at the [Card](https://github.com/aod/zhithead/blob/main/src/components/ui/Card.tsx) component and think I found it:

https://github.com/aod/zhithead/blob/7c786681b239984188d3c1ed4a48eccadcdaaf28/src/components/ui/Card.tsx#L36

The prop [layoutId](https://www.framer.com/motion/layout-animations/#shared-layout-animations) on a [motion component](https://www.framer.com/motion/introduction/##the-motion-component) enables this feature.
When you play a card it is removed from the [Hand](https://github.com/aod/zhithead/blob/main/src/components/ui/Hand.tsx) component, re-rendered without that card and rendered instead in the [Pile](https://github.com/aod/zhithead/blob/main/src/components/Pile.tsx) component.
I don't know the technical details on how Framer Motion transitions the element. But I do know that using that prop animates out the old component and transitions it to the new component's place.

Let me know if that answers your question. Good luck on your poker game!

**Reply by zhibin-wang09** [reply]
WOW, it worked. I remember I was reading your code and was tinkering with the animate prop for the <motion> component because I thought I have to manipulate the x and y coordinates for the card. In hindsight I literally missed it by one line😭. Thank You!
